handleMessage.go

```diff
@@ -91,12 +89,10 @@ func (srv *server) registerHandler(
 // deregisterHandler decrements the number of currently executed handlers
 // and shuts down the server if scheduled and no more operations are left
 func (srv *server) deregisterHandler(con *connection) {
-	srv.opsLock.Lock()
-	srv.currentOps--
-	if srv.shutdown && srv.currentOps < 1 {
+	srv.decOps()
+	if srv.isStopping() && srv.getOps() < 1 {
```

Comment:
Not sure whether this is acceptable, because, as you can see, `srv.currentOps` and `srv.shutdown` are used as one atomic entity protected by `srv.opsLock`. I'm not sure if they could get out of sync, but if they do then it could cause the server to not shut down even if instructed to.